internal/plugin/connectors/tcp/proxy_service.go

```diff
@@ -103,6 +103,7 @@ func (proxy *proxyService) handleConnection(clientConn net.Conn) error {
 	}()
 
 	backendCredentials, err := proxy.retrieveCredentials()
+	// zeroize credentials if we exit early due to an error
 	defer internal.ZeroizeCredentials(backendCredentials)
```

Comment:
not a blocker but this line should probably move below the error (or move into the cred retrieval) unless we return non-nil backendCredentials when error is set

Reply:
Without the `defer` statement located before the error returns, it is possible to still access the connection details, which may not happen outside of a unit test, but it's probably better to be safe and keep the defer statement where it is.